From a3a6763d0e656d62b10b570ead5fa7c77e2f2600 Mon Sep 17 00:00:00 2001
From: Karl Williamson <khw@cpan.org>
Date: Sun, 9 Nov 2025 13:41:59 -0700
Subject: [PATCH 02/10] locale.c: Silence unused variable compiler warning

On some systems this was unused.  Now that we have C99, we can move the
declaration and some #ifdef's and not declare it unless it is going to
be used.

(cherry picked from commit 323d85c86c657bb81983b0057d2310c52642797d)
---
 locale.c | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/locale.c b/locale.c
index 4fbefb1910..dda5e10e8c 100644
--- a/locale.c
+++ b/locale.c
@@ -8352,20 +8352,22 @@ S_ints_to_tm(pTHX_ struct tm * mytm,
     mytm->tm_year = year;
 
     struct tm * which_tm = mytm;
-    struct tm aux_tm;
 
 #ifndef HAS_MKTIME
 
     mini_mktime(mytm);
 
 #else
+#  if defined(HAS_TM_TM_GMTOFF) || defined(HAS_TM_TM_ZONE)
+#    define ALWAYS_RUN_MKTIME
+
+    struct tm aux_tm;
+
+#  endif
 
     /* On platforms that have either of these two fields, we have to run the
      * libc mktime() in order to set them, as mini_mktime() doesn't deal with
      * them.  [perl #18238] */
-#  if defined(HAS_TM_TM_GMTOFF) || defined(HAS_TM_TM_ZONE)
-#    define ALWAYS_RUN_MKTIME
-#  endif
 
     /* When isdst is 0, it means to consider daylight savings time as never
      * being in effect.  Many libc implementations of mktime() do not allow
-- 
2.53.0

